File: gitlfs-common/src/test/java/ru/bozaro/gitlfs/common/data/CreateLockResTest.java
package ru.bozaro.gitlfs.common.data;

import com.fasterxml.jackson.databind.util.StdDateFormat;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.io.IOException;
import java.text.ParseException;

public final class CreateLockResTest {
  @Test
  public void parse01() throws IOException, ParseException {
    final CreateLockRes data = SerializeTester.deserialize("create-lock-res-01.json", CreateLockRes.class);
    Assert.assertNotNull(data);

    final Lock lock = data.getLock();
    Assert.assertNotNull(lock);
    Assert.assertEquals(lock.getId(), "some-uuid");
    Assert.assertEquals(lock.getPath(), "/path/to/file");
    Assert.assertEquals(lock.getLockedAt(), StdDateFormat.instance.parse("2016-05-17T15:49:06+00:00"));
    Assert.assertNotNull(lock.getOwner());
    Assert.assertEquals(lock.getOwner().getName(), "Jane Doe");
  }
}

File: gitlfs-common/src/test/kotlin/ru/bozaro/gitlfs/common/data/CreateLockResTest.kt
package ru.bozaro.gitlfs.common.data

import com.fasterxml.jackson.databind.util.StdDateFormat
import org.testng.Assert
import org.testng.annotations.Test
import java.io.IOException
import java.text.ParseException

class CreateLockResTest {
    @Test
    @Throws(IOException::class, ParseException::class)
    fun parse01() {
        val data: CreateLockRes = SerializeTester.deserialize("create-lock-res-01.json", CreateLockRes::class.java)
        Assert.assertNotNull(data)
        val lock = data.lock
        Assert.assertNotNull(lock)
        Assert.assertEquals(lock.id, "some-uuid")
        Assert.assertEquals(lock.path, "/path/to/file")
        Assert.assertEquals(lock.lockedAt, StdDateFormat.instance.parse("2016-05-17T15:49:06+00:00"))
        Assert.assertNotNull(lock.owner)
        Assert.assertEquals(lock.owner!!.name, "Jane Doe")
    }
}
